src/parse.rs

```diff
@@ -123,6 +124,10 @@ impl Filter {
                 Some(right) => return Err(unexpected_argument("slugify", right)),
                 None => FilterType::Slugify(SlugifyFilter),
             },
+            "upper" => match right {
+                Some(right) => return Err(unexpected_argument("upper", right)),
```

Comment:
We also need a test for this branch. I'd write something similar to https://github.com/LilyFoote/django-rusty-templates/blob/eae574e935435f7e6c26352b45a71f0142f37812/tests/filters/test_escape.py#L17-L36